### ISSUE TITLE
Stranger Things with Epoxy + Animatable2Compat

### ISSUE THREAD

**Issue by bernaferrari** [airbnb/epoxy#718]
This is the weirdest issue I've ever had with Epoxy. I tried to triple-check everything, and it still looks like there is something wrong with Epoxy or Android.

Basically.. I have this:

```kotlin
(imageView.drawable as? Animatable2Compat)?.onAnimationEnd {
            if (currentlyReloading) (imageView.drawable as? Animatable)?.start()
}
```

When sync ends, `currentlyReloading` is set to false and the animation ends - without repeating... Or so I thought when I used Groupie. Because that code doesn't work at all with Epoxy, `currentlyReloading` var is true forever even if I set it false. I have absolutely no idea what is going on, looks like scheduler/synchronization issue, except I'm not changing the value at multiple places at once. I even put a custom set attribute to help detect if there something else was changing the value and nope.. it really becomes false, but for onAnimEnd it is still true.

This is the simplified project, you need to find `SampleKotlinModelWithHolder` and `startAndReloadAnim` inside of it. Even changing `currentlyReloading` to other variables (like `status == SYNC.LOADING` which is similar) produces the same effect.

TL;DR I HAVE NO IDEA WHAT IS GOING ON.

This is what is suppose to happen:
![GIF](https://raw.githubusercontent.com/bernaferrari/ChangeDetection/master/.github/assets/card_gif.gif)

What is happening: the "last sync" and the progress never ever stops spinning because condition never stops being satisfied because of mystery.

The simplified project is here:
[ChangeDetection.zip](https://github.com/airbnb/epoxy/files/3025718/ChangeDetection.zip)

My open source app which I'm migrating to Epoxy/MvRx is here https://github.com/bernaferrari/ChangeDetection/

**Comment by elihart** [comment]
This is almost certainly a bug with your code - I would not be so quick to blame Epoxy.

I did take a look at your code though and my guess is that your issue is `unregisterAnimationCallback` is never called. Every time a model is bound to the view it adds its on animation end listener, so many listeners are being called every time the animation ends, and one of those is restarting the animation.

**Comment by bernaferrari** [comment]
> This is almost certainly a bug with your code

You are right, sorry 😢. It was working fine with Groupie, and I would never think of that issue you mentioned when I tried to convert it. It was kind of unintuitive to know that I should `clearAnimationCallbacks` on bind, instead of unbind.

But yaaaaaaay!! Everything is working great again! Thanks a lot!!! My app will be able to live with Epoxy 😃😃😃!!
